File: 3582_generate_tag_for_video_caption/solution.cpp
// LeetCode 3582 - Generate Tag for Video Caption
// https://leetcode.com/problems/generate-tag-for-video-caption/

#include <cctype>
#include <sstream>
#include <string>

class Solution {
public:
    std::string generateTag(std::string caption) {
        std::istringstream iss(caption);
        std::string word, ans = "#";
        int i = 0;
        while (iss >> word) {
            for (char& c : word) c = (char)std::tolower((unsigned char)c);
            if (i == 0) ans += word;
            else {
                if (!word.empty()) word[0] = (char)std::toupper((unsigned char)word[0]);
                ans += word;
            }
            if ((int)ans.size() >= 100) break;
            i++;
        }
        if ((int)ans.size() > 100) ans.resize(100);
        return ans;
    }
};

File: 3582_generate_tag_for_video_caption/solution.php
<?php
// LeetCode 3582 - Generate Tag for Video Caption
// https://leetcode.com/problems/generate-tag-for-video-caption/

class Solution {
    function generateTag($caption) {
        $ans = '#';
        $words = preg_split('/\s+/', trim($caption));
        $i = 0;
        foreach ($words as $word) {
            if ($word === '') continue;
            $w = strtolower($word);
            if ($i === 0) $ans .= $w;
            else {
                if (strlen($w) > 0) $w = strtoupper($w[0]) . substr($w, 1);
                $ans .= $w;
            }
            if (strlen($ans) >= 100) break;
            $i++;
        }
        if (strlen($ans) > 100) $ans = substr($ans, 0, 100);
        return $ans;
    }
}
